apps/web/modules/ee/contacts/lib/contact-attribute-keys.ts

```diff
@@ -9,3 +11,92 @@ export const getContactAttributeKeys = reactCache(
     });
   }
 );
+
+export const getContactAttributeKeyById = async (
+  id: string
+): Promise<Pick<TContactAttributeKey, "id" | "environmentId" | "type" | "name" | "description"> | null> => {
+  const key = await prisma.contactAttributeKey.findUnique({
+    where: { id },
+    select: { id: true, environmentId: true, type: true, name: true, description: true },
+  });
+
+  return key;
+};
+
+export const createContactAttributeKey = async (data: {
+  environmentId: string;
+  key: string;
+  name?: string;
+  description?: string;
+}): Promise<TContactAttributeKey> => {
+  try {
+    const contactAttributeKey = await prisma.contactAttributeKey.create({
+      data: {
+        key: data.key,
+        name: data.name ?? data.key,
+        description: data.description ?? null,
+        environmentId: data.environmentId,
+        type: "custom",
+      },
+    });
+
+    return contactAttributeKey;
+  } catch (error) {
+    if (error instanceof Error && "code" in error) {
+      if (error.code === PrismaErrorType.UniqueConstraintViolation) {
+        throw new DatabaseError("Attribute key already exists");
```

Comment:
We also have a similar issue here with the creation as well, since this service is called directly from the action and no error is returned, the user loses this error. When I try to create a new attribute with an already existing key, I only see this from the UI:

<img width="802" height="766" alt="Image" src="https://github.com/user-attachments/assets/8b772f59-53e8-4203-b1a0-103b270776f6" />